.github/workflows/golangci-lint.yaml

```diff
@@ -0,0 +1,40 @@
+---
+name: golangci-lint
+
+on:
+  pull_request:
+    branches: ['*']
+
+jobs:
+  golangci-lint:
+    name: golangci-lint
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+      - name: Setup go env
+        uses: red-hat-storage/ocs-osd-deployer@main
+      - name: golangci-lint
+        uses: golangci/golangci-lint-action@v3
+        with:
+          # Optional: version of golangci-lint to use in form of v1.2 or v1.2.3 or `latest` to use the latest version
+          version: v1.46.2
+
+          # Optional: working directory, useful for monorepos
+          # working-directory: somedir
+
+          # Optional: golangci-lint command line arguments.
+          args: --timeout 3m --verbose
+
+          # Optional: show only new issues if it's a pull request. The default value is `false`.
+          # only-new-issues: true
+
+          # Optional: if set to true then the all caching functionality will be complete disabled,
+          #           takes precedence over all other caching options.
+          # skip-cache: true
+
+          # Optional: if set to true then the action don't cache or restore ~/go/pkg.
+          # skip-pkg-cache: true
+
+          # Optional: if set to true then the action don't cache or restore ~/.cache/go-build.
+          # skip-build-cache: true
```

Comment:
Why do we need all of these comments?

Reply:
We don't need those comments, it is just for reference in case we need to enable something.
It is kind of a template that golangci-lint provides, we can remove it.

Reply:
Please remove them, an up to date reference can be found online